src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -306,6 +307,30 @@ public void AddHostObject(string name, FormulaType type, Func<IServiceProvider,
                 displayName: default);
 
             _variables.Add(hostDName, info);
+        }
+
+        /// <summary>
+        /// Adds a UserInfo object schema.
+        /// Actual object is added in Runtime config service provider.
+        /// </summary>
+        public void AddUserInfoObject()
+        {
+            var userInfoType = RecordType.Empty()
+                .Add("FullName", FormulaType.String)
+                .Add("Email", FormulaType.String);
+
+            AddHostObject("UserInfo", userInfoType, GetUserInfoObject);
+        }
+
+        private FormulaValue GetUserInfoObject(IServiceProvider serviceProvider)
+        {
+            var userInfo = (IUserInfo)serviceProvider.GetService(typeof(IUserInfo)) ?? throw new Exception("UserInfo object was not added to service");

```

Comment:
>Exception [](http://example.com/codeflow?start=97&length=9)

C# tip - we should never throw "Exception" directly - instead throw an appropriate derived type.  InvalidOperationException is a good catch all. 


Although in this case, we could just return null and update the caller to convert that into an error. 

see [https://github.com/microsoft/Power-Fx/pull/1196](https://github.com/microsoft/Power-Fx/pull/1196)

Reply:
Actually, thinking more ... exception (instead of ErrorValue) is the right failure here since this scenario is a host bug. The host told us to enable User object, but then didn't provide the value for it. 

So just make it a more specific exception, like InvalidOperationException

Reply:
updated to InvalidOperationException